### PR TITLE
Extended syntax: remove unnamed alternatives from syntax

### PR DESCRIPTION
**Pull request by Anabra**

### CODE REVIEW

File: grin/src/Grin/ExtendedSyntax/Parse/AST.hs [hunk 1]
```diff
@@ -31,25 +31,26 @@ def = Def <$> try (L.indentGuard sc EQ pos1 *> var) <*> many var <* op "=" <*> (
...
-  t <- (L.indentGuard sc GT i >>= expr)
-  L.indentGuard sc EQ i
-  kw "else"
-  e <- (L.indentGuard sc GT i >>= expr)
-  return $ ECase b [ Alt (LitPat (LBool True))  t
-                   , Alt (LitPat (LBool False)) e
-                   ]
+-- NOTE: This becomes deprecated with the introduction of named alternatives.
+-- TODO: Repalce ifs with case expressions in code examples.
+-- ifThenElse :: Pos -> Parser Exp
```
Comment (andorp) [inline grin/src/Grin/ExtendedSyntax/Parse/AST.hs]: I think this change should be in its own PR with all the changes in the GRIN examples.

Comment (Anabra) [reply grin/src/Grin/ExtendedSyntax/Parse/AST.hs]: This was my first thought as well, but then I realized those code examples don't exist yet. The tests containing them haven't been copied and updated to the new syntax. For those that have been updated, `if`s have been removed (eg.: tests for some of the analyses).

Comment (andorp) [reply grin/src/Grin/ExtendedSyntax/Parse/AST.hs]: Than I would suggest to remove it completely. In general it is not a good idea to aggregate commented out code snippets.
